components/scream/scripts/test_all_scream.py

```diff
@@ -45,9 +47,32 @@ def __init__(self, cxx, kokkos, submit, parallel, fast_fail, baseline_ref, basel
                        "Requested test '{}' is not supported by test-all-scream, please choose from: {}".\
                            format(t, ", ".join(self._test_full_names.keys())))
 
-        if not self._baseline_dir == "NONE":
-            print ("Ignoring baseline ref {}, and using baselines in directory {} instead".format(self._baseline_ref, self._baseline_dir))
-            print ("NOTE: baselines for each build type BT must be in '{}/BT/data'. We don't check this, but there will be errors if the baselines are not found.".format(self._baseline_dir))
+        expect(self._src_dir.endswith("components/scream"), "Run from $scream_repo/components/scream")
+        if not self._kokkos:
+            expect(self._machine, "If no kokkos provided, must provide machine name for internal kokkos build")
+        if self._submit:
+            expect(self._machine, "If dashboard submit request, must provide machine name")
+
+        # Compute baseline info
+        import pdb
+        pdb.set_trace()
+        expect(not (self._baseline_ref and self._baseline_dir),
+               "Makes no sense to specify a baseline generation commit if using pre-existing baselines ")
+        if self._baseline_dir is None:
+            if self._baseline_ref is None:
+                # Compute baseline ref
+                if self._keep_tree:
+                    self._baseline_ref = "HEAD"
+                else:
+                    self._baseline_ref = get_common_ancestor("origin/master")
```

Comment:
This could be problematic. Say a user does not specify the baseline ref, and his/her branched forked from origin/master. The tests may pass, but that does not mean it's ok to merge. Even though the AT will test against master, and even though we should always specify the baseline ref, I find it safer to default to current origin/master, rather than the common ancestor. If you want to test against the common ancestor, you can figure out the SHA manually, then pass it to `-b`.

Reply:
If the user forks from commit A, produces commit B, which is BFB with A, but in the meantime, commit C has been added to origin/master which is not BFB, why would we want to fail the test?